Misc fixes for GD32

gmarull opened the pull request:
Found while working with https://github.com/zephyrproject-rtos/zephyr/pull/67124

gmarull committed:
modules: hal_gigadevice: use CONFIG_${ARCH} to select components

Instead of using custom SoC definitions. The selected components (e.g.
CMSIS), depend on the architecture selected by the SoC Kconfig options.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
drivers: counter: gd32: depend on !SOC_SERIES_GD32VF103

Because SOC_FAMILY_GD32_ARM is going to be dropped soon.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
boards: gd32f450i_eval: remove unused Kconfig

Board no longer has init code.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
boards: gd32f403z_eval: remove redundant Kconfig

Board does not have any init code, so BOARD_INIT_PRIORITY is redundant.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

nandojve commented on drivers/counter/Kconfig.gd32:
```diff
@@ -6,7 +6,7 @@
 	bool "GD32 timer counter driver"
 	default y
-	depends on DT_HAS_GD_GD32_TIMER_ENABLED && SOC_FAMILY_GD32_ARM
+	depends on DT_HAS_GD_GD32_TIMER_ENABLED
```
Is this compatible with RISC-V ?

CC: @soburi

soburi replied:
The timer has not yet worked with GD32V.
Please keep the restriction for ARM only.

soburi commented on drivers/counter/Kconfig.gd32:
```diff
@@ -6,7 +6,7 @@
 config COUNTER_TIMER_GD32
 	bool "GD32 timer counter driver"
 	default y
-	depends on DT_HAS_GD_GD32_TIMER_ENABLED && SOC_FAMILY_GD32_ARM
```
The timer has not yet worked with GD32V.
Please keep the restriction for ARM only

gmarull replied:
Does that mean GD32V has a different IP? If so, GD32V dts files should have another compatible, ie, not `gd,gd32-timer`. For now I'll change to `!SOC_SERIES_GD32VF103`, because `SOC_FAMILY_GD32_ARM` is gonna disappear with hwmv2.

soburi replied:
The timer driver depends on interrupt controller behavior. It works correctly on ARM, but it needs improvement for RISC-V.
It's still not going well.

nandojve replied:
Hi @soburi ,

Would you mind to open an issue and describe exactly what is the problem and the possible root of cause?

soburi replied:
I've got it. please wait a moment.

kartben commented:
@gmarull You may want to look into converting images to WebP. You can likely get similar visual quality for far less than 95K. 

You may use `cwebp`, and perhaps not use a fixed target size but rather a `q`uality level of the default one doesn't already get you a small enough size. This comment also applies to JPEG BTW: forcing a 95K target might lead to "too good" (i.e. unnecessarily under compressed, just to fit the target size) of an image.

gmarull committed:
boards: gd32*: reduce board image file size

Some GD32 board pictures exceed the maximum allowed limit (100K), so
reduce them.

Automated using `cwebp path/to/board.jpg -o path/to/board.webp`

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull commented:
@soburi ping

soburi commented:
@gmarull 

> @soburi ping

It may not be very organized, but I've created an issue.
https://github.com/zephyrproject-rtos/zephyr/issues/67425

gmarull commented:
> @gmarull
> 
> > @soburi ping
> 
> It may not be very organized, but I've created an issue. #67425

can you approve the PR in the mean time?

soburi commented:
> > @gmarull
> > > @soburi ping
> > 
> > 
> > It may not be very organized, but I've created an issue. #67425
> 
> can you approve the PR in the mean time?

Did it. I am sorry to have kept you waiting.
